core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java

```diff
@@ -0,0 +1,317 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.proxy.s3;
+
+import alluxio.client.file.URIStatus;
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * Get bucket result defined in http://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html
+ * It will be encoded into an XML string to be returned as an error response for the REST call.
+ *
+ * TODO(chaomin): consider add more required fields in S3 BucketGet API.
+ */
+@JacksonXmlRootElement(localName = "ListBucketResult")
+@JsonPropertyOrder(
+    { "Name", "Prefix", "KeyCount", "MaxKeys", "IsTruncated", "Contents" })
+public class ListBucketResult {
```

Comment:
Instead of making s3 compatible objects, could we use the s3 objects themselves?

Reply:
good point, I'm open to that. For now using our own for some customization, since we have some optional param that Alluxio does not support yet.

Reply:
I see, the benefit of using their object would be the "type safe" compliance, but if that is trivial I think the flexibility of our own type is also helpful.